### PR TITLE
BL-313 Created GET/instructor/profile/profile_id route

### PR DESCRIPTION
**Pull request by cavazosgeorge**
## Description

We built out the instructor/profile/:profile_id endpoint that allows us to request instructors based on their specifc profile_ids. 

Collaborators: [Berenika Ahmetaj](https://github.com/berenika14), [Anthony Amaya](https://github.com/aamay022)

Fixes # (issue)

We created a new model function `findInstructorProfileByProfileId` that sends us back the entire instructor profile object. 

## Loom Video

https://www.loom.com/share/be92709f5df244c58014046ffd1f00e7

## Type of change

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)
- [x] This change requires a documentation update

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have removed unnecessary comments/console logs from my code
- [x] I have made corresponding changes to the documentation if necessary (optional)
- [x] My changes generate no new warnings
- [x] I have checked my code and corrected any misspellings
- [x] No duplicate code left within changed files
- [x] Size of pull request kept to a minimum
- [x] Pull request description clearly describes changes made & motivations for said changes

### CODE REVIEW

**Review by JorgeEvangelista500 — APPROVED** [review]
Great work! everything works as intended, a great addition moving forward, to be able to get a specific instructor.

**Review by peterdavidconley — APPROVED** [review]
Great job on clean and simple code for "findInstructorProfileByProfileId". .Then & .Catch statements look flawless. No further critiques from me.

**Review by paulstgermain — CHANGES_REQUESTED** [review]
This PR needs a Loom recording walking us through your work before I can officially review it! 😱 

Please record a short Loom walking us through the work you did here and add a link to it in your description as soon as you can.

File: api/instructor/instructorRouter.js [hunk 1]
```diff
@@ -16,6 +16,18 @@ router.get('/courses', authRequired, getInstructorId, (req, res, next) => {
...
+router.get('/profile/:profile_id', authRequired, (req, res) => {
+  Instructors.findInstructorProfileByProfileId(req.params.profile_id)
+    .then((instructor) => {
+      if (instructor) {
+        res.status(200).json(instructor);
+      } else {
+        res.status(404).json({ error: 'Instructor not found.' });
+      }
+    })
+    .catch(errorhandler);
```
Comment (EduarddoEscobar) [inline api/instructor/instructorRouter.js]: you can change the .catch(errorhandler) to .catch(next) to stay more consistent with the other endpoint.

File: README.md [hunk 2]
```diff
@@ -82,9 +83,10 @@
 }
 ```
 
-| Method | URL                      | Description                                                         |
-| ------ | ------------------------ | ------------------------------------------------------------------- |
-| [GET]  | /instructor/courses/ | Returns an array containing all course event objects belonging to the currently logged in instructor. |
+| Method | URL                            | Description                                                                                           |
+| ------ | ------------------------------ | ----------------------------------------------------------------------------------------------------- |
+| [GET]  | /instructor/courses/           | Returns an array containing all course event objects belonging to the currently logged in instructor. |
+| [GET]  | /instructor/profile/profile_id | Returns a profile object of a specific instructor .                                                   |
```
Comment (EduarddoEscobar) [inline README.md]: Changing /instructor/profile/profile_id to /instructor/profile/:profile_id could help more quickly understand that you need to pass the profile_id for the endpoint

**Review by EduarddoEscobar — APPROVED** [review]
Overall good work just some slight changes to stay more consistent

File: api/instructor/instructorRouter.js (hunk 1 above)
Comment (Berenika14) [reply api/instructor/instructorRouter.js]: Hey, thanks for catching that, used your recommendation and looks much more consistent and cleaner with the rest of the code.

**Comment by Berenika14** [comment]
> This PR needs a Loom recording walking us through your work before I can officially review it! 😱
> 
> Please record a short Loom walking us through the work you did here and add a link to it in your description as soon as you can.


Hey, @paulstgermain Loom video added. 🙂

**Review by Ike888 — APPROVED** [review]
Great work done here! and some ideas for the Auth0 as well.

**Review by kylerostler — APPROVED** [review]
Great job with this, love how useful you made the endpoint for getting the entire instructor profile object. Looks like this needs a quick Loom though so we can see what changes were made!

**Review by jose-mercado — APPROVED** [review]
Nice job creating an endpoint to fetch an instructor based on a profile ID! Looks like there are some conflicts in the README and api/instructor/instructorRouter.js, but aside from that, this looks great!

File: api/instructor/instructorRouter.js [hunk 3]
```diff
@@ -16,6 +16,18 @@ router.get('/courses', authRequired, getInstructorId, (req, res, next) => {
     .catch(next);
 });
 
+router.get('/profile/:profile_id', authRequired, (req, res, next) => {
```
Comment (dirkknibbe) [inline api/instructor/instructorRouter.js:19]: Good job using the same format for functions that come before, very readable.

**Review by dirkknibbe — APPROVED** [review]

**Review by attaris978 — APPROVED** [review]
Your code is minimalistic, efficient, and easy to read. Very well done.
More emphatically, I absolutely adore your attention to display detail with the README.md file. You took something that was annoying to follow and made it visually appealing and eminently more readable.
The only improvement I can think of is to be careful of punctuation parallel structure -- some of your Description column entries use periods, others do not -- though this is hardly a substantive issue.

**Pull request closed without merging**